src/features/Admin/Rooms/AddNewRoom/AddNewRoom.tsx

```diff
@@ -1,9 +1,407 @@
-import React from 'react'
+import { Button, Checkbox, Divider, FormControl, Grid, ListItemText, Paper, Select, TextField } from "@mui/material";
+import { Box, Container } from "@mui/system";
+import { useContext, useState, useEffect } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import Styles from "./AddNewRoom.module.scss";
+import { InputLabel, MenuItem } from "@mui/material";
+import axios from "axios";
+import { AuthContext } from "../../../../context/AuthContext";
+import { addroomsUrl } from "../../../../services/api";
+import { IAddRoom } from "../../../../interface/RoomInterface";
+import { faciRoomsUrl } from "./../../../../services/api";
+import DragDropFileUpload from "../../../../shared/DragDropFileUpload/DragDropFileUpload";
+const AddNewRoom: React.FC = () => {
+  const { requestHeaders } = useContext(AuthContext);
+
+  const navigate = useNavigate();
+
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [facilitiesList, setFacilitiesList] = useState([]);
+  const [selectedImage, setSelectedImage] = useState(null);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    watch,
+    formState: { errors },
+
+  } = useForm<IAddRoom>();
+
+  // Format Data
+  const appendToFormData = (data: IAddRoom) => {
+    const formData = new FormData();
+    formData.append("roomNumber", data["roomNumber"]);
+    formData.append("price", data["price"]);
+    formData.append("capacity", data["capacity"]);
+    formData.append("discount", data["discount"]);
+    // formData.append("facilities", data["facilities"]?.[0]);
+
+    if (Array.isArray(data.facilities)) {
+      data.facilities.forEach((facility) => {
+        formData.append("facilities[]", facility);
+      });
+    } else if (data.facilities) {
```

Comment:
I'm unclear about the necessity of the second condition. Could you please provide some clarification on its purpose? Additionally, it would be beneficial to include a comment for it, elaborating on the rationale behind it. Thank you!

Reply:
In the beginning, I didn't notice that facilities are multi-option so when running code for the first time message in "network" "facilities must be arrayed" when searching in Google about this message and asking AI chat gives me this "If data. facilities is not an array but still has a value, we assume it is a single facility and append it to the form data" means that if data. facilities is not an array, it is treated as a single facility value" When I knew in the Sunday session that it multi-select forgot to edit this